tests/unittest/llmapi/test_serve_report_addr.py

```
@@ -0,0 +1,179 @@
+    timer = threading.Timer(1.0, lambda: _publish_bound_address(addr_path, "localhost", 12345))
+    timer.start()
+    try:
+        assert wait_for_reported_addr(addr_path, timeout=20, process=_FakeProcess()) == (
+            "localhost",
+            12345,
+        )
+    finally:
+        timer.join()
+
+
+@pytest.mark.parametrize("port,report_addr", [(0, "/tmp/a.addr"), (8000, "/tmp/a.addr"), (0, None)])
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/a.addr"

(S108)